packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -8,55 +8,92 @@ export default class ScheduledPaymentsFetcherService {
-              },
-              scheduledPaymentAttempts: {
-                none: {
-                  status: 'inProgress',
-                },
-              },
-            },
-          },
-        ],
-      },
-    });
+    let results: [{ id: string; started_at: Date }] = await prisma.$queryRaw`SELECT 
+        scheduled_payments.id AS id,
+        last_failed_payment_attempt.started_at
+      FROM 
+        scheduled_payments
+      LEFT JOIN (SELECT scheduled_payment_attempts.scheduled_payment_id,
+                  MAX(scheduled_payment_attempts.started_at) AS started_at
+            FROM scheduled_payment_attempts
+            WHERE scheduled_payment_attempts.status = 'failed'
+            GROUP BY scheduled_payment_id) AS last_failed_payment_attempt
+      ON last_failed_payment_attempt.scheduled_payment_id = scheduled_payments.id
+      WHERE 
+        (
+          scheduled_payments.canceled_at IS NULL 
+          AND scheduled_payments.creation_block_number > 0 
+          AND scheduled_payments.pay_at > ${startOfDay(subDays(nowUtc, this.validForDays))}
+          AND scheduled_payments.pay_at <= ${nowUtc}
+          AND (
+            (
+              scheduled_payments.recurring_day_of_month IS NULL 
+              AND (
+                scheduled_payments.id
+              ) NOT IN (
+                SELECT 
+                  scheduled_payments1.id 
+                FROM 
+                  scheduled_payments AS scheduled_payments1 
+                  INNER JOIN scheduled_payment_attempts AS scheduled_payment_attempts1 
+                  ON (scheduled_payment_attempts1.scheduled_payment_id) = (scheduled_payments1.id) 
+                WHERE 
+                  (
+                    (
+                      scheduled_payment_attempts1.status = 'succeeded'
+                      OR scheduled_payment_attempts1.status = 'in_progress'
+                    ) 
+                    AND scheduled_payments1.id IS NOT NULL
+                  )
+              )
+            ) 
+            OR (
+              scheduled_payments.recurring_day_of_month > 0 
+              AND scheduled_payments.recurring_until >= ${nowUtc}
+              AND (
+                scheduled_payments.id
+              ) NOT IN (
+                SELECT 
+                  scheduled_payments1.id 
+                FROM 
+                  scheduled_payments AS scheduled_payments1 
+                  INNER JOIN scheduled_payment_attempts AS scheduled_payment_attempts1 
+                  ON (scheduled_payment_attempts1.scheduled_payment_id) = (scheduled_payments1.id) 
+                WHERE 
+                  (
+                    scheduled_payment_attempts1.status = 'in_progress'
+                    AND scheduled_payments1.id IS NOT NULL
+                  )
+              )
+            )
+          )
+        )
+        ORDER BY 
+          CASE WHEN last_failed_payment_attempt.started_at IS NULL 
+          THEN ${convertDateToUTC(new Date(0))} 
```

Comment:
You can use `utcNow()`. But I'm not sure I understand why sorting by an evaluated date value. Shouldn't this be sorting on a database field?

Please add ASC/DESC to make it clearer.

Reply:
I'm not trying to get the current date time. What I'm trying to do is insert the oldest date time into the scheduled payment with no payment attempts. So I can sort scheduled payments by started_at of the last failed payment attempt.

Reply:
(@jurgenwerk `new Date(0)` is jan 1st, 1970)

Reply:
is this so that nulls are sorted at the end? i think that there is a command you can use to tell postgres to place nulls at the end (or the beginning). i vaguely recall having to do this in the past

Reply:
Here it is:
```
ORDER BY somevalue DESC NULLS LAST
```
https://stackoverflow.com/questions/7621205/sort-null-values-to-the-end-of-a-table